src/app/components/shared/date-time-filter/filter.component.spec.ts

```diff
@@ -0,0 +1,535 @@
+      updateForm();
+
+      // assert that the correct error message is shown to the user
+      expect(getElementByInnerText(outOfBoundsDateErrorMessage)).toExist();
+    }));
+
+    it("should not show an error if the user inputs a valid date range", fakeAsync(() => {
+      const startDate = "2019-12-10";
+      const endDate = "2020-10-10";
+
+      spectator.typeInElement(startDate, getDateStartedAfterInput());
+      spectator.typeInElement(endDate, getDateFinishedBeforeInput());
+      updateForm();
+
+      // assert that the correct error message is shown to the user
+      expect(getElementByInnerText(outOfBoundsDateErrorMessage)).not.toExist();
+    }));
+
+    it("should not show an out of bounds date error if the user has only input a start date", fakeAsync(() => {
+      const startDate = "2021-10-12";
+
+      spectator.typeInElement(startDate, getDateStartedAfterInput());
+      updateForm();
+
+      expect(getElementByInnerText(outOfBoundsDateErrorMessage)).not.toExist();
+    }));
+
+    it("should not show an out of bounds date error if the user has only input an end date", fakeAsync(() => {
+      const endDate = "2021-09-01";
+
+      spectator.typeInElement(endDate, getDateFinishedBeforeInput());
+      updateForm();
+
+      expect(getElementByInnerText(outOfBoundsDateErrorMessage)).not.toExist();
+    }));
+
+    it("should not show an out of bounds date error if the user has not input any dates", fakeAsync(() => {
+      updateForm();
+      expect(getElementByInnerText(outOfBoundsDateErrorMessage)).not.toExist();
+    }));
+  });
+
+  describe("time bounds validations", () => {
+    const outOfBoundsTimeErrorMessage =
+      // eslint-disable-next-line max-len
+      "Time of day filtering does not currently support filtering past a day boundary. Please adjust your request so that the start time is before the end time.";
+
+    beforeEach(
+      fakeAsync(() => toggleTodFilters())
+    );
+
+    [true, false].forEach((dstValue: boolean) => {
+      it(`should not show an error message if the time bounds are valid and dst is set to ${dstValue}`, fakeAsync(() => {
+        const startTime = "10:00";
+        const endTime = "11:15";
+
+        spectator.typeInElement(startTime, getTodStartedAfterInput());
+        spectator.typeInElement(endTime, getTodFinishedBeforeInput());
+        updateForm();
+
+        expect(getElementByInnerText(outOfBoundsTimeErrorMessage)).not.toExist();
+      }));
+
+      it(`should show an error message if the time bounds are not valid and dst is set to ${dstValue}`, fakeAsync(() => {
+        const startTime = "15:12";
+        const endTime = "12:11";
+
+        spectator.typeInElement(startTime, getTodStartedAfterInput());
+        spectator.typeInElement(endTime, getTodFinishedBeforeInput());
+        updateForm();
+
+        expect(getElementByInnerText(outOfBoundsTimeErrorMessage)).toExist();
+      }));
+    });
+  });
+
+  describe("filters events", () => {
```

Comment:
make most of these filter tests unit tests...

Perhaps even move filter creation out into it's own class?